### PR TITLE
CAMS-326 Prevent duplicate transfers

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Group CTO transactions from DXTR for a given case ID into one transfer order. This results in one or more docket entries being shown for a given transfer order.

# Major Changes

* Refactored the transfer order model to allow for multiple docket entries.
* Refactored models to be implemented and imported from common from API and UI.
* Modified the TransferOrderAccordion to show one or more docket entries in details.

# Testing/Validation

* Fixed tests broken by the model refactor.
* Manual testing that the case duplication is resolved.

### CODE REVIEW

File: common/src/cams/orders.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  id: string;
+  caseId: string;
+  status: 'rejected';
+  reason?: string;
+};
+
+type TransferOrderActionApproval = {
+  id: string;
+  caseId: string;
+  newCase?: Partial<CaseSummary>; // TODO CASM-326 may want to make this required
```
Comment (btposey) [inline common/src/cams/orders.ts:31]: Let see if requiring this breaks anything...

File: common/src/cams/orders.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { CaseDetailInterface, CaseDocketEntry, CaseSummary } from './cases';
+
+export type OrderStatus = 'pending' | 'approved' | 'rejected';
+export type OrderType = 'transfer' | 'consolidation';
+
+// TODO: Consider modeling a CaseDetailSummary in cases.d.ts and use it here instead of CaseDetailInterface
```
Comment (btposey) [inline common/src/cams/orders.ts:6]: Stale note.

File: user-interface/src/lib/type-declarations/chapter-15.d.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { ResponseData } from './api';
+import * as CommonOrders from '@common/cams/orders';
 
 export type OrderStatus = 'pending' | 'approved' | 'rejected';
```
Comment (btposey) [inline user-interface/src/lib/type-declarations/chapter-15.d.ts:4]: This is in common. We should export it from there.

File: backend/functions/lib/adapters/gateways/dxtr/orders.dxtr.gateway.ts [hunk 4]
```diff
@@ -46,36 +52,81 @@ export class DxtrOrdersGateway implements OrdersGateway {
         value: txId,
       });
 
-      const rawOrders = await this._getOrders(context, txId, chapters, regions);
-      context.logger.info(MODULE_NAME, `Retrieved ${rawOrders.length} raw orders from DXTR.`);
-      const documents = await this._getDocuments(context, txId, chapters, regions);
+      // Get raw order which are a subset of case detail associated with a transfer order
+      const rawOrders = await this._getTransferOrders(context, txId, chapters, regions);
```
[the remark this answers is not in the document]
Comment (btposey) [reply backend/functions/lib/adapters/gateways/dxtr/orders.dxtr.gateway.ts:56]: Implied they are for private use. Should use a `private` keyword on them.

File: backend/functions/lib/adapters/gateways/orders.cosmosdb.repository.test.ts [hunk 5]
```diff
@@ -146,7 +153,7 @@ describe('Test case assignment cosmosdb repository tests', () => {
   });
 
   test('When putting an order, Should throw ServerConfigError if an AggregateAuthenticationError error occurs', async () => {
-    const testNewOrderData2: Order = {
+    const testNewOrderData2: TransferOrder = {
```
Comment (amorrow-flexion) [inline backend/functions/lib/adapters/gateways/orders.cosmosdb.repository.test.ts:156]: Not a fan of naming like this.

**Review by fmaddenflx — APPROVED** [review]
Looks good.

File: backend/functions/lib/adapters/gateways/dxtr/orders.dxtr.gateway.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fmaddenflx) [reply backend/functions/lib/adapters/gateways/dxtr/orders.dxtr.gateway.ts:56]: Thought so.  Yes.  Should use private.

**Pull request merged**